library/Makefile

```diff
@@ -18,6 +18,9 @@ endif
 # To compile on Plan9:
 # CFLAGS += -D_BSD_EXTENSION
 
+# Generate dependency information on headers in GNU Make format
+DEPFLAGS = -MT $@ -MMD -MP -MF $*.d
```

Comment:
These flags only work for GCC, there are similer ones for other compilers e.g. ARMCC and IAR